tests/r8.rs

```diff
@@ -98,3 +99,42 @@ fn test_uuid_win() {
         "d8b03b44-58df-5cd7-adc7-aefcfb0e2ade".parse().unwrap()
     );
 }
+
+#[test]
+fn test_remap_source_file() {
+    let mapping = ProguardMapping::new(MAPPING_R8_SYMBOLICATED_FILE_NAMES);
+
+    let mapper = ProguardMapper::new(mapping);
+
+    let test = mapper.remap_stacktrace(r#"
+    Caused by: java.lang.Exception: Hello from main!
+	at a.a.a(SourceFile:12)
+	at io.wzieba.r8fullmoderenamessources.MainActivity.b(SourceFile:6)
+	at io.wzieba.r8fullmoderenamessources.MainActivity.a(SourceFile:1)
+	at a.c.onClick(SourceFile:1)
+	at android.view.View.performClick(View.java:7659)
+	at android.view.View.performClickInternal(View.java:7636)
+	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)"#
+    );
+
+    /**
+    Ideally, should be (output from `retrace` tool):
+    Caused by: java.lang.Exception: Hello from main!
+   	at io.wzieba.r8fullmoderenamessources.Foobar.foo(Foobar.java:10)
```

Comment:
Thanks for the vigilance, but it's not a copy-paste - `retrace` generates `Foobar.java`. I don't know why though 😕

Reply:
Maybe `retrace` itself does not honor these comments :-D
I believe this is an `r8` extension that might not be supported in `retrace` at all.

Reply:
> Maybe retrace itself does not honor these comments :-D

You mean it doesn't look at `#" {"id":"sourceFile","fileName":"...)` and just takes the file name from the class name and adds `.java` suffix? Yeah, it might be the case. It's a pity that there's no source code to peek 😕 . 

I think the solution we work on in this PR, based on  `#" {"id":"sourceFile","fileName":"...)` whenever we can is a solid one though and more bulletproof. I'd vote to keep file names (e.g. `.kt` here instead of `.java`) as it's the truth (even if out of sync with what `retrace` returns).

Reply:
Actually, the source code of retrace is available I think here: https://r8.googlesource.com/r8/+/f0816ebb923855c8946ed8008629e6a73f345ad3/src/main/java/com/android/tools/r8/retrace/

I didn't have yet a chance to investigate it though

Reply:
https://r8.googlesource.com/r8/+/f0816ebb923855c8946ed8008629e6a73f345ad3/src/main/java/com/android/tools/r8/retrace/internal/RetraceUtils.java#74 looks like it's happening here

Reply:
Thanks! It looks like it should support persisting `.kt` extension. The data I used for testing, that resulted changing `.kt` to `.java` is:

<details><summary>stacktrace</summary>

```
    Caused by: java.lang.Exception: Hello from main!
	at a.a.a(SourceFile:12)
	at io.wzieba.r8fullmoderenamessources.MainActivity.b(SourceFile:6)
	at io.wzieba.r8fullmoderenamessources.MainActivity.a(SourceFile:1)
	at a.c.onClick(SourceFile:1)
	at android.view.View.performClick(View.java:7659)
	at android.view.View.performClickInternal(View.java:7636)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
```

</details>

<details><summary>mapping file</summary>

```
# compiler: R8
# compiler_version: 8.3.36
# min_api: 24
# common_typos_disable
# {"id":"com.android.tools.r8.mapping","version":"2.2"}
# pg_map_id: 38f963d
# pg_map_hash: SHA-256 38f963d5fa0b2339093e05fb7a90056de20b885ec51b8d11204e67f10ec1f670
io.wzieba.r8fullmoderenamessources.Foobar -> a.a:
# {"id":"sourceFile","fileName":"Foobar.kt"}
    1:7:void foo():9:9 -> a
    8:15:void foo():10:10 -> a
io.wzieba.r8fullmoderenamessources.FoobarKt -> a.b:
# {"id":"sourceFile","fileName":"Foobar.kt"}
    1:5:void main():15:15 -> a
    6:9:void main():16:16 -> a
    1:4:void main(java.lang.String[]):0:0 -> b
io.wzieba.r8fullmoderenamessources.MainActivity -> io.wzieba.r8fullmoderenamessources.MainActivity:
# {"id":"sourceFile","fileName":"MainActivity.kt"}
    1:4:void <init>():7:7 -> <init>
    1:4:void $r8$lambda$pOQDVg57r6gG0-DzwbGf17BfNbs(android.view.View):0:0 -> a
      # {"id":"com.android.tools.r8.synthesized"}
    1:9:void onCreate$lambda$1$lambda$0(android.view.View):14:14 -> b
    1:3:void onCreate(android.os.Bundle):10:10 -> onCreate
    4:8:void onCreate(android.os.Bundle):12:12 -> onCreate
    9:16:void onCreate(android.os.Bundle):13:13 -> onCreate
    17:20:void onCreate(android.os.Bundle):12:12 -> onCreate
io.wzieba.r8fullmoderenamessources.MainActivity$$ExternalSyntheticLambda0 -> a.c:
# {"id":"sourceFile","fileName":"R8$$SyntheticClass"}
# {"id":"com.android.tools.r8.synthesized"}
    1:4:void onClick(android.view.View):0:0 -> onClick
      # {"id":"com.android.tools.r8.synthesized"}
io.wzieba.r8fullmoderenamessources.R -> a.d:
    void <init>() -> <init>
      # {"id":"com.android.tools.r8.synthesized"}
```

</details>

and command as simple as `retrace mapping.txt stacktrace.txt`.

<details><summary>output</summary>

```
Caused by: java.lang.Exception: Hello from main!
	at io.wzieba.r8fullmoderenamessources.Foobar.foo(Foobar.java:10)
	at io.wzieba.r8fullmoderenamessources.MainActivity.onCreate$lambda$1$lambda$0(MainActivity.java:14)
	at io.wzieba.r8fullmoderenamessources.MainActivity.$r8$lambda$pOQDVg57r6gG0-DzwbGf17BfNbs(MainActivity.java:0)
	at io.wzieba.r8fullmoderenamessources.MainActivity$$ExternalSyntheticLambda0.onClick(MainActivity.java:0)
	at android.view.View.performClick(View.java:7659)
	at android.view.View.performClickInternal(View.java:7636)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
```

</details>


Sharing in case you'd like to validate - maybe my local instance of `retrace` is out of date or something.

Anyway, I think that keeping original extension (`.kt` in this case) is the way to go.